pkg/cluster/operation/destroy.go

```diff
@@ -308,19 +308,33 @@ func CleanupComponent(getter ExecutorGetter, instances []spec.Instance, cls spec
 		log.Infof("Cleanup instance %s", ins.GetHost())
 
 		delFiles := set.NewStringSet()
+		dataPaths := set.NewStringSet()
+		logPaths := set.NewStringSet()
 
 		if options.CleanupData && len(ins.DataDir()) > 0 {
 			for _, dataDir := range strings.Split(ins.DataDir(), ",") {
-				delFiles.Insert(path.Join(dataDir, "*"))
+				dataPaths.Insert(path.Join(dataDir, "*"))
 			}
 		}
 
 		if options.CleanupLog && len(ins.LogDir()) > 0 {
 			for _, logDir := range strings.Split(ins.LogDir(), ",") {
-				delFiles.Insert(path.Join(logDir, "*"))
+				logPaths.Insert(path.Join(logDir, "*.log"))
 			}
 		}
 
+		for _, lp := range logPaths.Slice() {
+			for _, dp := range dataPaths.Slice() {
+				// remove path of log dir from list if any data dir is under it
+				if strings.HasPrefix(dp, lp) {
```

Comment:
I don't think this will work because the logPaths is like:

```
/deploy/log/*.log
```

And the data dir could be:

```
/deploy/log/data
```

In this case, the data dir is under log dir, but it's not detected.